poly-synth/src/poly-synth-demo.js

```diff
@@ -0,0 +1,100 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * @class PolySynthDemo
+ * @description Triggers synthesizer events based on GUI input.
+ */
+class PolySynthDemo {
+  /**
+   * @constructor
+   * @param {AudioContext} context The audio context.
+   */
+  constructor(context) {
+    this.masterGain_ = new GainNode(context, {gain: 0.5});
+    this.polySynth_ = new PolySynth(context);
+    this.polySynth_.output.connect(this.masterGain_)
+        .connect(context.destination);
+
+    // Initialize a QWERTY keyboard defined in qwerty-hancock.min.js.
+    // The key A on a computer keyboard maps onto the note C, and the letters
+    // to the right of A (SDFGHJ) map onto the notes above C in a C major scale
+    // (DEFGAB). The letters WETYU map onto the black keys of a piano.
+    this.keyboard_ = new QwertyHancock(
+        {id: 'keyboard', width: 600, height: 150, octaves: 2});
+
+    this.keyboard_.keyDown = this.keyDown.bind(this);
+    this.keyboard_.keyUp = this.keyUp.bind(this);
+  }
+
+  /**
+   * Initialize GUI components.
+   * @param {String} containerId the id of the HTML container
+   */
+  initializeGUI(containerId) {
+    let lowPassSlider_ = new ParamController(
+        containerId, this.polySynth_.setParameter.bind(this.polySynth_), {
+          type: 'range',
+          min: this.polySynth_.lowPassMinCutoff,
+          max: this.polySynth_.lowPassMaxCutoff,
+          step: 1,
+          default: this.polySynth_.lowPassMaxCutoff,
+          id: 'cutoff',
+          name: 'Cutoff'
+        });
+
+    // The maximum gain is set to be larger than 1 to allow for highly filtered
+    // sounds to be audible. It is up to the user to avoid distortion.
+    let masterGainSlider_ =
+        new ParamController(containerId, this.setGain.bind(this), {
+          type: 'range',
+          min: 0,
+          max: 5,
+          step: 0.01,
+          default: this.masterGain_.gain.value,
+          id: 'masterGain',
+          name: 'Volume'
+        });
+  }
+
+  /**
+   * Change the volume.
+   * This is bound to an event listener by a ParamController.
+   * @param {Number} value the new gain
+   */
+  setGain(value) {
+    this.masterGain_.gain.value = parseFloat(value);
+  }
+
+  /**
+   * Play a note when a mapped key is pressed. This method implements a function
+   * triggered in qwerty-hancock.min.js.
+   * @param {String} note the note to be played, e.g. A4 for an octave four A
+   * @param {Number} pitch the corresponding pitch of the note, e.g 440
```

Comment:
I'm not sure I can do it in one line of code since I need to know the relative order of the notes. It's a bit more complicated but happy to add it. It's just more code.